From 0b35fb5980f5ed79bad4417816a2d5079095b109 Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Sun, 16 Jun 2019 14:46:48 +0300
Subject: [PATCH 293/386] arm64: dts: cn9132: add cn9132-db device tree with
 setup B

The cn913x DB board is a modular board,
which supports multiple possible configurations.

Topology for setup B:
AP-MPPs:
	- SDIO, UART
CP0-SerDes:
	- lane0: PCIe0
	- lane1: PCIe0
	- lane2: PCIe0
	- lane3: PCIe0
	- lane4: SFI0
	- lane5: SATA1
CP0-MPPs:
	- RGMII1, RGMII2, SDIO, I2C, NAND
CP1-SerDes:
	- lane0: PCIe0
	- lane1: PCIe0
	- lane2: SFI0
	- lane3: USB3_1
	- lane4: SGMII1
	- lane5: SATA1
CP1-MPPs:
	- I2C, SPI1
CP2-SerDes:
        - lane0: PCIe0
        - lane1: PCIe0
        - lane2: SATA0
        - lane3: USB3_1
        - lane4: SFI0
        - lane5: PCIe2
CP2-MPPs:
        - SDIO

Change-Id: I761cc794538476703fbc6de65ec42c6ed7a49828
Signed-off-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/10979
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Marcin Wojtas <marcin@marvell.com>
Reviewed-by: Stefan Chulski <Stefan.Chulski@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 .../bindings/arm/marvell/cn913x.txt           |  4 +
 arch/arm64/boot/dts/marvell/Makefile          |  1 +
 arch/arm64/boot/dts/marvell/cn9132-db-B.dts   | 76 +++++++++++++++++++
 3 files changed, 81 insertions(+)
 create mode 100644 arch/arm64/boot/dts/marvell/cn9132-db-B.dts

diff --git a/Documentation/devicetree/bindings/arm/marvell/cn913x.txt b/Documentation/devicetree/bindings/arm/marvell/cn913x.txt
index 54ea6e349976..97eaa754e9e5 100644
--- a/Documentation/devicetree/bindings/arm/marvell/cn913x.txt
+++ b/Documentation/devicetree/bindings/arm/marvell/cn913x.txt
@@ -37,6 +37,10 @@ Boards:
       when the board being used is the OcteonTX2 CN9132 Development board with
       Setup A.
 
+   - "marvell,cn9132-db-B"
+      when the board being used is the OcteonTX2 CN9132 Development board with
+      Setup B.
+
 Example:
 
 compatible = "marvell,cn9131-db-A", "marvell,cn9131-db",
diff --git a/arch/arm64/boot/dts/marvell/Makefile b/arch/arm64/boot/dts/marvell/Makefile
index 3b51c90c02f8..110e3bb65d19 100644
--- a/arch/arm64/boot/dts/marvell/Makefile
+++ b/arch/arm64/boot/dts/marvell/Makefile
@@ -30,5 +30,6 @@ dtb-$(CONFIG_ARCH_MVEBU) += cn9130-db-B.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += cn9131-db-A.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += cn9131-db-B.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += cn9132-db-A.dtb
+dtb-$(CONFIG_ARCH_MVEBU) += cn9132-db-B.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += cn9130-crb-A.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += cn9130-crb-B.dtb
diff --git a/arch/arm64/boot/dts/marvell/cn9132-db-B.dts b/arch/arm64/boot/dts/marvell/cn9132-db-B.dts
new file mode 100644
index 000000000000..fe0c706dddee
--- /dev/null
+++ b/arch/arm64/boot/dts/marvell/cn9132-db-B.dts
@@ -0,0 +1,76 @@
+/*
+ * copyright (c) 2019 marvell international ltd.
+ *
+ * spdx-license-identifier:    gpl-2.0
+ * https://spdx.org/licenses
+ */
+
+#include "cn9131-db-B.dts"
+#include "cn9132-db.dtsi"
+
+/ {
+	model = "Model: Marvell CN9132 development board (CP NAND) setup(B)";
+	compatible = "marvell,cn9132-db-B", "marvell,armada-ap807-quad",
+		     "marvell,armada-ap807";
+
+	aliases {
+		gpio5 = &cp2_gpio1;
+		gpio6 = &cp2_gpio2;
+		ethernet5 = &cp2_eth0;
+	};
+};
+
+&cp2_ethernet {
+	status = "okay";
+};
+
+/* SLM-1521-V2, CON9 */
+&cp2_eth0 {
+	status = "okay";
+	phy-mode = "10gbase-kr";
+	/* Generic PHY, providing serdes lanes */
+	phys = <&cp2_comphy4 0>;
+	managed = "in-band-status";
+	sfp = <&cp2_sfp_eth0>;
+};
+
+/* SLM-1521-V2, CON6 */
+&cp2_pcie0 {
+	status = "okay";
+	num-lanes = <2>;
+	num-viewport = <8>;
+	/* Generic PHY, providing serdes lanes */
+	phys = <&cp2_comphy0 0
+		&cp2_comphy1 0>;
+};
+
+/* SLM-1521-V2, CON8 */
+&cp2_pcie2 {
+	status = "okay";
+	num-lanes = <1>;
+	num-viewport = <8>;
+	/* Generic PHY, providing serdes lanes */
+	phys = <&cp2_comphy5 2>;
+};
+
+&cp2_sata0 {
+	status = "okay";
+	/* SLM-1521-V2, CON4 */
+	sata-port@0 {
+		status = "okay";
+		/* Generic PHY, providing serdes lanes */
+		phys = <&cp2_comphy2 0>;
+	};
+};
+
+/* CON 2 on SLM-1683 - microSD */
+&cp2_sdhci0 {
+	status = "okay";
+};
+
+/* SLM-1521-V2, CON11 */
+&cp2_usb3_1 {
+	status = "okay";
+	/* Generic PHY, providing serdes lanes */
+	phys = <&cp2_comphy3 1>;
+};
-- 
2.17.1

